api/storage_service.cc

```diff
@@ -668,21 +668,14 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto keyspace = validate_keyspace(ctx, req->param);
         auto table_infos = parse_table_infos(keyspace, ctx, req->query_parameters, "cf");
         apilog.debug("force_keyspace_compaction: keyspace={} tables={}", keyspace, table_infos);
+
+        auto& cm = db.local().get_compaction_manager();
+        auto compaction_module_ptr = cm.get_compaction_module().shared_from_this();
+        auto task_impl_ptr = std::make_unique<major_keyspace_compaction_task_impl>(compaction_module_ptr, keyspace, db, table_infos);
```

Comment:
keyspace and table_infos can be moved i to the task impl, no?

Reply:
Not really due to that:
https://github.com/scylladb/scylladb/blob/6a2ef15365a2a066c02f38f7913d739657ba21fc/api/storage_service.cc#L680
Keyspace can be moved and then got from task, but table_infos should stay as is I think.